### ISSUE TITLE
FTransUNet：训练时间太长的问题

### ISSUE THREAD

**Issue by wuwangabc** [sstary/SSRS#65]
在训练FTransUNet时，训练时间太长了,几分钟才几个batch,我查看了我使用了gpu,我使用的是笔记本4060,想问一下作者有什么问题吗？

**Comment by sstary** [comment]
你检查下你是否正确调用了GPU。正常情况下，最多几分钟就会打印一次，即迭代了100个batch。
也可以换个模型试一下，排除是模型问题，还是环境问题。

**Comment by wuwangabc** [comment]
好的

**Comment by wuwangabc** [comment]
我已检查，正确调用了gpu,是否我的算力过低，导致训练过慢

**Comment by sstary** [comment]
其余超参都和我github一致吗？4060应该不会很慢的。
你说的几个batch，是真的几个，还是打印log上的几个？log是100个batch 打印一次的。

**Comment by wuwangabc** [comment]
参数没有改动，这是train函数代码def train(net, optimizer, epochs, scheduler=None, weights=WEIGHTS, save_epoch=1):
    losses = np.zeros(1000000)
    mean_losses = np.zeros(100000000)
    weights = weights.cuda()

    criterion = nn.NLLLoss(weight=weights)
    iter_ = 0
    acc_best = 90.0

    for e in range(1, epochs + 1):
        if scheduler is not None:
            scheduler.step()
        net.train()
        for batch_idx, (data, dsm, target) in enumerate(train_loader):
            # 打印当前学习率
            current_lr = optimizer.param_groups[0]['lr']
            print(f"Epoch {e}, Batch {batch_idx}, Current Learning Rate: {current_lr}")
            data, dsm, target = Variable(data.cuda()), Variable(dsm.cuda()), Variable(target.cuda())
            optimizer.zero_grad()
            output = net(data, dsm)
            loss = CrossEntropy2d(output, target, weight=weights)
            loss.backward()
            optimizer.step()

            losses[iter_] = loss.data
            mean_losses[iter_] = np.mean(losses[max(0, iter_ - 100):iter_])

            if iter_ % 100 == 0:
                clear_output()
                rgb = np.asarray(255 * np.transpose(data.data.cpu().numpy()[0], (1, 2, 0)), dtype='uint8')
                pred = np.argmax(output.data.cpu().numpy()[0], axis=0)
                gt = target.data.cpu().numpy()[0]
                print('Train (epoch {}/{}) [{}/{} ({:.0f}%)]\tLoss: {:.6f}\tAccuracy: {}'.format(
                    e, epochs, batch_idx, len(train_loader),
                    100. * batch_idx / len(train_loader), loss.data, accuracy(pred, gt)))
            iter_ += 1
            del (data, target, loss)

            # if e % save_epoch == 0:
            if iter_ % 500 == 0:
                net.eval()
                acc = test(net, test_ids, all=False, stride=Stride_Size)
                net.train()
                if acc > acc_best:
                    torch.save(net.state_dict(), './resultsv_se_ablation/segnet256_epoch{}_{}'.format(e, acc))
                    acc_best = acc
    print('acc_best: ', acc_best)我在其中加一句打印batch  ---------   print(f"Epoch {e}, Batch {batch_idx}")

**Comment by wuwangabc** [comment]
参数未改动，我在train函数中加入一句打印batch--------print(f"Epoch {e}, Batch {batch_idx}")
该位置如下
def train(net, optimizer, epochs, scheduler=None, weights=WEIGHTS, save_epoch=1):
    losses = np.zeros(1000000)
    mean_losses = np.zeros(100000000)
    weights = weights.cuda()

    criterion = nn.NLLLoss(weight=weights)
    iter_ = 0
    acc_best = 90.0

    for e in range(1, epochs + 1):
        if scheduler is not None:
            scheduler.step()
        net.train()
        for batch_idx, (data, dsm, target) in enumerate(train_loader):
            # 打印batch# 打印batch
            print(f"Epoch {e}, Batch {batch_idx}")
            # 打印batch
            data, dsm, target = Variable(data.cuda()), Variable(dsm.cuda()), Variable(target.cuda())
            optimizer.zero_grad()
            output = net(data, dsm)
            loss = CrossEntropy2d(output, target, weight=weights)
            loss.backward()
            optimizer.step()

            losses[iter_] = loss.data
            mean_losses[iter_] = np.mean(losses[max(0, iter_ - 100):iter_])

            if iter_ % 100 == 0:
                clear_output()
                rgb = np.asarray(255 * np.transpose(data.data.cpu().numpy()[0], (1, 2, 0)), dtype='uint8')
                pred = np.argmax(output.data.cpu().numpy()[0], axis=0)
                gt = target.data.cpu().numpy()[0]
                print('Train (epoch {}/{}) [{}/{} ({:.0f}%)]\tLoss: {:.6f}\tAccuracy: {}'.format(
                    e, epochs, batch_idx, len(train_loader),
                    100. * batch_idx / len(train_loader), loss.data, accuracy(pred, gt)))
            iter_ += 1
            del (data, target, loss)

            # if e % save_epoch == 0:
            if iter_ % 500 == 0:
                net.eval()
                acc = test(net, test_ids, all=False, stride=Stride_Size)
                net.train()
                if acc > acc_best:
                    torch.save(net.state_dict(), './resultsv_se_ablation/segnet256_epoch{}_{}'.format(e, acc))
                    acc_best = acc
    print('acc_best: ', acc_best)

**Comment by wuwangabc** [comment]
![image](https://github.com/user-attachments/assets/7af65686-38b3-4ee0-91c9-f7cceea3d1c7)
差不多6分钟过去了，batch就这么多

**Comment by wuwangabc** [comment]
好的

**Comment by wuwangabc** [comment]
没有，我只运行了这一个程序，改动也只是在路径上有时候他会默认加上//，这样就和\不匹配找不到路径，所以只是在路径连接后面加上\，我晚上再跑一下其他模型吧

**Comment by wuwangabc** [comment]
作者，我在重新运行一遍，差不多100个batch要运行13分钟。作者用的是什么GPU呢？

**Comment by sstary** [comment]
我3090和4090都用过，但是不会你这么慢，我是linux系统。你试试其他模型
